### PR TITLE
Mag cal values

### PR DESCRIPTION
**Pull request by ctzsnooze**
## For testing only!

This PR makes two changes to the Sensors Tab display of Mag data.

- Uses 'raw' Mag units in the display of Mag sensor data, using PR [3593](https://github.com/betaflight/betaflight-configurator/pull/3593)

- For each axis, shows, to the right of the instantaneous Mag sensor value, the average of the min and max values received on that axis since the Sensors Tab was opened.  This value is preceded by 'C' for 'Cal`.

If the user goes `set mag_calibration = 0,0,0` and clears their existing cal values, they can move the quad around so that, sequentially, one Mag axis, and then its opposite end, is pointed North into the local magnetic field vector.  When a max and min have been defined, the value after C is the cal value to use for the axis. 

Also a previous calibration can be checked.  A good calibration will return within 1% of zero on each axis.  This can be used to validate the 'hard iron' or 'zero point' calibration that Betaflight uses.

I also calculate the root (sum of squares of x,y,x) value, into `magSumSqR` which can be thought of as the 'radius' of the 'sphere' of mag readings.  The value should equal the local value of the earths field, after correction for the internal gain of the sensor.  Furthermore, if there are no soft iron effects, the value should remain constant regardless of the angle of the quad.  If the 'sphere' appears deformed, the 'radius' will not be consistent, and the value will decrease, or increase, consistently, at certain angles.  BUT... I can't figure out how to display this `magSumSqR` value; if anyone could help me with that, I'd be grateful.

I made a small video temporarily here...

https://github.com/betaflight/betaflight-configurator/assets/11737748/a7348ee1-5d20-4e6b-886d-94d6809dd8d4

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/tabs/sensors.js
Comment (chmelevskij) [inline src/js/tabs/sensors.js:470]: Do we still need these comments here?

**Review by chmelevskij — APPROVED** [review]

File: src/js/tabs/sensors.js
Comment (ctzsnooze) [reply src/js/tabs/sensors.js:470]: no, not really, they can go away.  Just kept them to remember the original code.  
I'll remove them when the consensus is to merge or not merge.
What we really need is a dedicated 'Mag' tab, similar in concept to the dedicated 'GPS' tab, where we can perform calibrations, check raw values, etc.

**Comment by ctzsnooze** [comment]
@chmelevskij @nerdCopter @haslinghuis 

How could I display the 'root sum of squares' values?  I would be grateful for a code suggestion.  

I would be happy to display it, and indeed the min, max and cal values, in a separate html element underneath the mag trace, rather than messing around with the raw sensor values, something like this:

```
field strength: <root sum of squares>
X axis : <min> <max> <cal>
Y axis : <min> <max> <cal>
Z axis : <min> <max> <cal>
```

My problem is that I'm not a js coder.  I know how to insert a div element into the html under the waveform, but I don't know how to 'load' such an element with values like the above.  The values already exist in the .js code, I just don't know how to add them to the UI.  I would be very grateful if someone could explain how to do this, with clear instructions.

**Comment by chmelevskij** [comment]
@ctzsnooze where about do you want to show those?

**Comment by ctzsnooze** [comment]
Closing since the basics have been done in a different PR.

**Pull request closed without merging**
